runners/spark/src/main/java/org/apache/beam/runners/spark/SparkRunner.java

```diff
@@ -120,13 +121,14 @@ private SparkRunner(SparkPipelineOptions options) {
     mOptions = options;
   }
 
-
   @Override
   public EvaluationResult run(Pipeline pipeline) {
     try {
       LOG.info("Executing pipeline using the SparkRunner.");
-
-      if (mOptions.isStreaming()) {
```

Comment:
Why not keep relying on options, and run the detecting visitor before this line ?
BTW, not sure why Flink named this optimizer... I think something like `StreamingPipelineDetector/StreamingModeDetector` is more fitting.